migration/20201112-add-incommon-saml-federation-metadata.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python
+"""Adds InCommon SAML federation metadata to `samlfederations` table."""
+
+import os
+import sys
+
+from contextlib2 import closing
+
+bin_dir = os.path.split(__file__)[0]
+package_dir = os.path.join(bin_dir, "..")
+sys.path.append(os.path.abspath(package_dir))
+
+from core.model import (
+    production_session
+)
+from api.saml.metadata.federations import incommon
+from api.saml.metadata.federations.model import SAMLFederation
+
+with closing(production_session()) as db:
+    incommon_federation = db.query(SAMLFederation).filter(
+        SAMLFederation.type == incommon.FEDERATION_TYPE).one_or_none()
+
+    if not incommon_federation:
+        incommon_federation = SAMLFederation(
+            incommon.FEDERATION_TYPE,
+            incommon.IDP_METADATA_SERVICE_URL,
+            incommon.CERTIFICATE
+        )
+
+        db.add(incommon_federation)
+        db.commit()
```

Comment:
This adds a `SAMLFederation` for sites that are being migrated from an earlier release. What about sites that are started up after this release happens? I don't think they'll have this.

I recommend a follow-up branch where you look at `SessionManager.initialize_data` and change it to use a registry, so different data model classes can register data initialization code.

Since data initialization code runs every time the server starts up, this will let you get rid of the migration script -- write the code correctly and a site will be guaranteed to have an appropriate SAMLFederation object on startup.